### ISSUE TITLE
Pop up no longer works with updated browser-mod

### ISSUE THREAD

**Issue by masterfish1** [fratsloos/fr24_card#19]
I imagine this is related to the update to browser_mod (https://github.com/thomasloven/hass-browser_mod) as I just had to edit all my personal popups but, as a result the popup in your card no longer works.

**Comment by fratsloos** [comment]
I see there is a change in the browser_mod. I'll look into it. Unfortunately I won't be able to do that for the next two or three weeks, because I can't access my development system untill then.

As a temporary workaround you can set the value for the popup to false:
```
type: custom:fr24-card
entity: sensor.fr24_aircraft
popup: false
```

**Comment by masterfish1** [comment]
No rush. I was just letting you know rather than requiring it be fixed immediately.

The workaround isn't needed as the card itself works absolutely fine even with "popup: true". It's just that the popup doesn't work when on true.

**Comment by fratsloos** [comment]
I've finally found some time to work on this issue. I had to completely rebuild my development system, which took some time. I fixed the issue with the popup itself, but it's not complete yet. When working with the color configuration options, the background of the popup is not completely filled. This is because of the many changes in the code of browser_mod. Will continue working on that. Just wanted to let you know that the issue is in progress. :)

**Comment by masterfish1** [comment]
Thanks for the update, I appreciate it.

Still no rush for me and actually less so at the moment as I'm moving house next week and my FR24 plane tracker is currently in a box.

**Comment by fratsloos** [comment]
@masterfish1 Are you completely settled after the move, and have you already taken HA and your plane tracker out of the box?

I made a pre-release of the card today and tested it in production at my place. Can you also test the release? If you have installed it via HACS you can choose 'show beta versions' at the redownload option. Then you can install version `0.6.0-alpha`.

**Comment by masterfish1** [comment]
Semi settled. Still loads to unpack but the plane tracker is done.

I won't get chance to test this until tomorrow afternoon but will do so and report back.

**Comment by fratsloos** [comment]
No problem. I'll wait for your feedback. Thanks for testing.

**Comment by BrettEBowman** [comment]
The pop-ups are working for me ok. One thing that I notice is that there isn't a "X" to close the pop-up which is sort of counter-intuitive. I think that it is because you don't have a "Title" object for the pop-up? I'm wondering if that main first line (with the flag, registration, and icon) could be the browser_mod pop-up window title (and then a "X" to close would be displayed?). Also, if the "Flight" is known and different from the "Registration", could that be added to that title line?

**Comment by masterfish1** [comment]
@fratsloos It isn't working for me but given that it is working for you and @BrettEBowman, I suspect the issue is with my install. Unfortunately, I don't have time to investigate right now.

Sorry for not getting back to you yesterday, Sunday was even worse than Saturday in the end.

**Comment by fratsloos** [comment]
@BrettEBowman

> One thing that I notice is that there isn't a "X" to close the pop-up which is sort of counter-intuitive. I think that it is because you don't have a "Title" object for the pop-up?

You're right. Because `title` was not used, the popup from browser_mod did not render a header with the cross to close the popup.
I modified this in the last commit by using the textual value of the title below the aircraft image as the popup title. For below the image I created a subtitle that now shows other data:
- if there is a registration: title contains registration, subtitle contains flight number or ICAO code
- if there is no registration, but there is a flight number: title contains flight number, subtitle the ICAO code
- if there is no registration and no flight number: both contain the ICAO code.

![Screenshot 2022-10-17 at 20 03 17](https://user-images.githubusercontent.com/9197177/196251636-cc5dfae1-1dc0-44a5-9e30-38585a3da0c8.png)

> (with the flag, registration, and icon)

That would indeed have been nice. Unfortunately it seems that the popup only accepts text as input and not HTML. [See the source code](https://github.com/thomasloven/hass-browser_mod/blob/master/js/plugin/popup-card.ts#L108).

@masterfish1 

> Sorry for not getting back to you yesterday, Sunday was even worse than Saturday in the end.

No problem! It's a hobby for all of us, right? ;)

Just to be sure, did you fully install the new version of browser_mod? I got in the sidebar of HA an item 'Browser Mod' in which I first had to turn on the 'Register' option.

**Comment by fratsloos** [comment]
I've just released version v0.6.0-beta which is available in HACS beta versions with the latest changes.

**Comment by masterfish1** [comment]
Seems to be working fine for me. Even the pop-up is working now. It started working on the alpha version too without me changing anything so no idea what was going on there at first.

**Comment by BrettEBowman** [comment]
I'm not seeing the Title in the pop-up. I'm pretty sure that the new code is in place OK because the "sub-title" line (with the flag and horizonal icon) is showing the new format.

Is there a change that I need to make to the card's configuration?

**Comment by BrettEBowman** [comment]
Update: I'm getting the Title in the pop-up when I use Edge as the browser, but not with Chrome. I'll have to try to dig into it more later and see if I can figure out why.

**Comment by fratsloos** [comment]
> EDIT: I got it to work in Chrome. I followed the instructions from the browser_mod page to force a hard cache reload. (I had already tried just doing a Ctrl-F5, which didn't make any difference.)

Yes, in these cases it's a caching issue almost every time. HA is known for an aggressive caching method. Just a tip, when `Ctrl`+`F5` doesn't do the trick, you can try it with `Ctrl`+`Shift`+`R`, or for the Mac users `⌘` + `Shift` + `R` ([docs](https://support.google.com/chrome/answer/157179?hl=en&co=GENIE.Platform%3DDesktop#zippy=%2Cwebpage-shortcuts)).

**Comment by fratsloos** [comment]
Fixed in [v0.6.0](https://github.com/fratsloos/fr24_card/releases/tag/v0.6.0).
